### ISSUE TITLE
Sever side errors are not handled and break

### ISSUE THREAD

**Issue by ScottSmix** [piceaTech/ember-rapid-forms#13]
I am using ember-rapid-forms 0.6.2 (released today). Comparing files, I have the same version of all the key dependencies (like ember-cli, ember, ember-data, ember-validations, etc).

When a server side validation fails, the current code does not seem to handled it and worse causes a runtime error (unless I've done something incorrectly). Take the simple controller below that is linked to a model that has the EV.Mixin:

```
import Ember from 'ember';

export default Ember.Controller.extend({
    actions: {
        save: function() {
            if (this.get('model.isValid')) {
                var _this = this;
                this.get('model').save().then(
                    function(friend) {
                        try {
                            _this.transitionToRoute('friends.show', friend);
                        }
                        catch(err) {
                            _this.set('errorMessage', 'Error: Internal error in route transition.');
                        }
                    },
                    function(err) {
                        _this.set('errorMessage', 'Error: ' + err.message );
                    }
                );
            } else {
                // should never happen because submit should be disabled when invalid
                this.set('errorMessage', 'Error: One or more input fields are not valid.');
            }
            return false;
        }
    }
});
```

When there is a server error, the this.get('model').save() promise rejects with an error: 

TypeError: recordErrors.add is not a function
    at Ember.Object.extend.adapterDidInvalidate (http://localhost:4200/assets/vendor.js:69415:24)
    at ember$data$lib$system$store$$Service.extend.recordWasInvalid 
...

Since there is always a chance that a server error may not be covered by the client, it seems like the server error should be returned instead.

Feedback and/or help would be much appreciated.

**Comment by ScottSmix** [comment]
Seems to be related to <a href="https://github.com/dockyard/ember-validations/issues/268">https://github.com/dockyard/ember-validations/issues/268</a>

I think ember rapid forms requires that ember validations be tied to the model (and not the controller). Currently with ember 1.11.x and associated ember-data, using ember-validations 2.0.0-alpha.3 breaks ember-data or break each other.

Of the work arounds proposed in the linked thread, any preferred approach for this project?

**Comment by pedrokiefer** [comment]
Not sure which of the work arounds is better, I like the idea of adding a prefix to `isValid`and `errors`, but that is a big API change, not sure if the ember-validations team is willing to do that.

We can add an integration test to track this issue, and others that might come up on one of our dependencies.

**Comment by ScottSmix** [comment]
I am using this work around in the meantime: I replaced node_modules/ember-validations/addon/validators/errors.js with the code below. Seems to work ok but need tests.

```
// in node_modules/ember-validations/addon/validators/errors.js
import Ember from 'ember';
import DS from 'ember-data';

export default DS.Errors.extend({
  unknownProperty: function(property) {
    this.set(property, Ember.makeArray());
    return this.get(property);
  }
});
```

**Comment by spruce** [comment]
I would like to wait a little longer until ember-validations sort out their problems. So that we don't fix things that can be fixed somewhere else

**Comment by spruce** [comment]
fixed as of merging of #141
